mcs/class/corlib/System.Reflection/ConstructorInfo.cs

```diff
@@ -1,146 +0,0 @@
-//
-// System.Reflection/ConstructorInfo.cs
-//
-// Author:
-//   Paolo Molaro (lupus@ximian.com)
-//
-// (C) 2001 Ximian, Inc.  http://www.ximian.com
-// Copyright (C) 2004-2005 Novell, Inc (http://www.novell.com)
-//
-// Permission is hereby granted, free of charge, to any person obtaining
-// a copy of this software and associated documentation files (the
-// "Software"), to deal in the Software without restriction, including
-// without limitation the rights to use, copy, modify, merge, publish,
-// distribute, sublicense, and/or sell copies of the Software, and to
-// permit persons to whom the Software is furnished to do so, subject to
-// the following conditions:
-// 
-// The above copyright notice and this permission notice shall be
-// included in all copies or substantial portions of the Software.
-// 
-// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
-// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
-// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
-// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
-// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
-// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
-// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
-//
-
-using System.Diagnostics;
-using System.Globalization;
-using System.Runtime.InteropServices;
-
-namespace System.Reflection {
-
-	[ComVisible (true)]
-	[ComDefaultInterfaceAttribute (typeof (_ConstructorInfo))]	
-	[Serializable]
-	[ClassInterface(ClassInterfaceType.None)]
-#if MOBILE
-	public abstract class ConstructorInfo : MethodBase {
-#else
-	public abstract class ConstructorInfo : MethodBase, _ConstructorInfo {
-#endif
-		[ComVisible (true)]
-		public static readonly string ConstructorName = ".ctor";
-		[ComVisible (true)]
-		public static readonly string TypeConstructorName = ".cctor";
-
-		protected ConstructorInfo() {
-		}
-		
-		[ComVisible (true)]
-		public override MemberTypes MemberType {
-			get {return MemberTypes.Constructor;}
-		}
-
-		[DebuggerStepThrough]
-		[DebuggerHidden]
-		public object Invoke (object[] parameters)
-		{
-			return Invoke (BindingFlags.CreateInstance, null, parameters ?? EmptyArray<object>.Value, null);
```

Comment:
Should CoreFX type also include this part?

Reply:
Brought it back in https://github.com/mono/corefx/pull/112 to fix some tests related invocation on abstract.